src/openrct2/world/Vec.hpp

```diff
@@ -0,0 +1,192 @@
+/*****************************************************************************
+ * Copyright (c) 2025 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "Location.hpp"
+
+#include <cmath>
+
+struct VecXY
```

Comment:
Why not use CoordsXY? It is basically a vector just different name.

Reply:
Hm right, I haven't really thought about that, I however honestly don't like that we have serval types doing the same thing essentially the only difference being the underlying type here. I wanted to have a base vector class for ages but there was push back and it got overruled to stick with the current code duplication, this however adds one more so its a bit of a weird situation.